src/util/states.ts

```diff
@@ -454,7 +457,8 @@ export function blurSeries(
             || blurScope === 'coordinateSystem' && !sameCoordSys
             // Not blur self series if focus is series.
             || focus === 'series' && sameSeries
-            // TODO blurScope: coordinate system
+            // Not blur other series if focus is self
+            || focus === 'self' && !sameSeries
```

Comment:
We still need to blur other series if `blurScope` is 'global' and `focus` is 'self'. You can run visual regression test(VRT in short) on `hoverFocus.html`, `hoverFocus2.html` to see if anythings breaks. The command of starting VRT is:
```ts
npm run test:visual
```

Reply:
@pissang Hi，I found it's complicated to first blur then highlight for each series, like this;
```javascript
for(const series of seriesList){
    blurNecessaryPart(series);
    highlightCurrent(series);
}
```

Since the previous highlighted series can be easily blurred by the latter series;
It may be more clear if we first blur all the necessary parts then hight all the necessary parts, like this;

```javascript
for(const series of seriesList){
    blurNecessaryPart(series);
}
for(const series of seriesList){
    highlightCurrent(series);
}
```

I modified the current code to this scenario and tested with `npm run test:visual`, please take a look~

Reply:
@ssthouse That's brilliant! It should work well. I can't find other issues. The only thing that needs to be done is record the user interactions on the new add test case so we can test it automatically.

You can click the `Record Interaction` on the VRT page button to open the recording page. Then follow the hint on the page and use the shortcut key to start recording. During the recording, you can click buttons, hover elements. Each click will trigger a screenshot to compare, but hover won't. So if you want to compare the current picture, you should at least click or use the shortcut key to trigger the screenshot. 

Usually, each action needs to focus on one test case and the interaction count in each action should be as less as possible. Feel free to ask me if you have problems on it.

Reply:
@pissang I have recorded the testcases and tried to run them. 
Currently, it failed(as expected), is there anything I should do to make it pass?

![image](https://user-images.githubusercontent.com/10973821/124249693-b24d1a80-db56-11eb-9769-b669352b2c60.png)